p4-16/psa/PSA.mdk

```diff
@@ -2572,3 +2543,131 @@ Certainly if there are long durations of time when egress decides to
 drop all packets to an output port, that port will go idle.  The
 scheduling algorithm implementations are all built with a finite
 maximum packet scheduling rate.
+
+
+# Appendix: Packet ordering {#appendix-packet-ordering}
+
+This section describes _recommendations_ for PSA implementations on
+the order that packets are processed.  These are not requirements,
+since there are known implementation techniques, especially
+parallelism that can be taken advantage of in a variety of ways, that
+can lead to cheaper implementations if these recommendations are not
+followed.  We recommend that developers selecting P4 devices ask their
+designers about these issues, if those developers consider the issues
+important for their purposes.
+
+Recommendation 1: Packets that arrive on the same input port should
+begin ingress processing in the same relative order as they arrived.
+
+Recommendation 2: Packets that go out on the same output port should
```

Comment:
Maybe the other way around, i.e. they should be transmitted in the same order they begin egress processing, but ultimately both are equivalent. For me it uis just easier to think from left to right :)

Reply:
I have rephrased Recommendation 2 for 'left to right' ordering :)